src/dataset/dataset_coco.py

```diff
@@ -312,3 +316,41 @@ def extract_patches(self, output_dir: str, patch_size: int, stride: int, min_are
                 pass
 
         patch_annotations.save(output_path=os.path.join(output_dir, "annotations", "annotations.json"))
+
+
+def extended_dimensions(patch_size: int, stride: int, image_shape: Tuple[int, ...]) -> Tuple[int, ...]:
+    # TODO test cases with strides greater than patch_size
+    return [
+        image_shape[i]
+        if (image_shape[i] - patch_size) % stride == 0
+        else (((image_shape[i] // stride) * stride) + patch_size)
+        for i in range(len(image_shape))
```

Comment:
What about breaking this statement into multiple lines to make it more readable? This is a tricky algorithm. Please comment the else block if you find it necessary.